### DISCUSSION TITLE
Add an audio cue for "Mute Myself"

### DISCUSSION THREAD

**Discussion by bspratt** [jamulussoftware/jamulus#1040] [UI Related]
The big red "MUTED" bar is a welcome visual cue, but even that isn't enough sometimes. We're not always looking at the client screen while performing, and that **_"oh sh*t I was on Mute Myself the whole time"_** experience is still sadly common. 

If the **client could inject a brief tone in place of the audio stream, say, once every five seconds when in "Mute Myself" mode** that might save some embarrassment and make the world a happier place. 

There must be cases in which this would NOT be desirable, so it should be a toggle in the client UI. The big red bar itself could be the button that toggles the audio cue:

![image](https://user-images.githubusercontent.com/1200761/94452244-2e4ba980-0164-11eb-84b9-c3ce2f839c1f.png)

Default value should be "off", to avoid nasty surprises for anyone using muted clients in piping schemes (e.g. I imagine WorldJam does so with remote studios). The revised text on the big red bar/button is advertisement enough for the new feature.

**Comment by chrisrimple** [comment]
An audio cue would be a great addition.  During the last World Jam, one musician thought he was muted and was practicing "behind" a live song, not realizing that his sound was in the mix.  With an audible cue, not hearing it would have reminded him that he wasn't muted.

Maybe include a slider or Volume Level for the audible cue, similar to New Client Level?

**Comment by bspratt** [comment]
I was thinking it was clever to repurpose the Big Red Bar real estate so as to not further crowd the UI. 

Perhaps where you click on the bar from left to right adjusts the tone's volume? But how to communicate that to the user is a question.

**Comment by chrisrimple** [comment]
Since the big red bar is pushing down all the other controls in the window, maybe change the big red bar to be something like...

MUTED (...)     [X] Audio cue  [-----o-----]

[X] is a checkbox for the audio cue
[-----o-----] is a horizontal slider "in" the big red bar

**Comment by bspratt** [comment]
Sounds like the right idea. 

Unfortunately none of this helps the **_"oh sh*t I was NOT muted the whole time"_** scenario you mentioned.... that's a separate issue, perhaps best titled **"pressing 'Mute' on your own slider should turn on 'Mute Myself'"**

**Comment by bspratt** [comment]
I think, though, that for a first go I'd just keep it simple and have the bar be a button with the tone at a fixed reasonably loud level.

**Comment by chrisrimple** [comment]
Yeah, maybe just 50% of server mix to start.  And true that it doesn't help the "I'm not muted after all problem", although perhaps the audible cue trains people to listen for it to ensure that they're muted.

**Comment by bspratt** [comment]
That's best entered as a new and independent issue.

**Comment by dingodoppelt** [comment]
Hi guys,
while I like the idea of an audible cue I have worked around the problem of not being able to see if you're muted by adding the mute status to the window title. This will be shown in the taskbar as well even when the jamulus window isn't visible.
![muteindicator](https://user-images.githubusercontent.com/62596379/95428679-a460c500-0949-11eb-902c-bf2737ce88b9.jpg)

You can find my changes here, if you're interested in trying it out:
https://github.com/dingodoppelt/jamulus/tree/muteindicator

**Comment by bspratt** [comment]
That seems like a really good parallel approach to the issue, thanks!

**Comment by gilgongo** [comment]
Hi - in an effort to tidy up the issues into a list of actionable work tickets, I'll move this to a discussion so that we can work on a spec if that's OK.

**Reply by bspratt** [reply]
This seems actionable enough, but I'm not going to make a fuss about it. Thanks for your efforts on the project.

**Reply by gilgongo** [reply]
Yes it was whether we want to do both audible and UI, or one or the other. @dingodoppelt has an example of the UI ony approach. If that's good enough then it's Ticket Time.

**Reply by bspratt** [reply]
The UI idea has its merits as an incremental improvement on the existing "Mute Myself" indicator, but doesn't actually solve the described problem: players typically are not looking at the UI while playing.  Decent idea, but a different ticket.
